storybookjs/storybook
# UI: Upgrade Icon component

## What I did

We introduced a new library for icons. In this PR I'm replacing all icons by the new library. This PR doesn't include the following:
- Icons that have dynamic names
- Icons with styling overrides
- Icons in the legacy component library. Not necessary.

This PR also fixes:
- Removed dependencies in the Storybook icons
- Update the versions of React to make sure it works with the versions of React supported by Storybook.

## How to test

1. Run a sandbox for template, e.g. `yarn task --task sandbox --start-from auto --template react-vite/default-ts`
2. Open Storybook in your browser
3. Access any story and check across addons and the manager if icons are not rendered properly.

## Checklist



      [MIGRATION.MD](https://github.com/storybookjs/storybook/blob/next/MIGRATION.md)

#### Maintainers


`["cleanup", "BREAKING CHANGE", "feature request", "bug", "build", "documentation", "maintenance", "dependencies", "other"]`



### 🦋 Canary release



This PR does not have a canary release associated. You can request a canary release of this pull request by mentioning the `@storybookjs/core` team here.

_core team members can create a canary release [here](https://github.com/storybookjs/storybook/actions/workflows/canary-release-pr.yml) or locally with `gh workflow run --repo storybookjs/storybook canary-release-pr.yml --field pr=<PR_NUMBER>`_
